django/db/models/query_utils.py

```diff
@@ -169,6 +173,94 @@ def _check_parent_chain(self, instance, name):
         return None
 
 
+class RegisterLookupMixin(object):
+    def _get_lookup(self, lookup_name):
+        try:
+            return self.class_lookups[lookup_name]
+        except KeyError:
+            # To allow for inheritance, check parent class' class_lookups.
+            for parent in inspect.getmro(self.__class__):
+                if 'class_lookups' not in parent.__dict__:
+                    continue
+                if lookup_name in parent.class_lookups:
+                    return parent.class_lookups[lookup_name]
+        except AttributeError:
+            # This class didn't have any class_lookups
+            pass
+        return None
+
+    def _get_transform(self, lookup_name):
+        # TODO(Josh): `_get_transform` and `_get_lookup` need to be combined somehow
+        try:
+            return self.class_transforms[lookup_name]
+        except KeyError:
+            # To allow for inheritance, check parent class' class_lookups.
+            for parent in inspect.getmro(self.__class__):
+                if 'class_transforms' not in parent.__dict__:
+                    continue
+                if lookup_name in parent.class_transforms:
+                    return parent.class_transforms[lookup_name]
+        except AttributeError:
+            # This class didn't have any class_transforms
+            pass
+        return None
+
+    def get_lookup(self, lookup_name):
+        from django.db.models.lookups import Lookup
```

Comment:
Because too many things depend on this class. `fields.Field`, `expressions.Func`, and `lookups.Lookup` all depend on this. In turn, `lookups` also depend on `expressions`, and `expressions` depend on `fields`.

`django.db.models` "conveniently" imports lots of the above, so simply importing `django.db.models` and then referencing `lookups.Lookup` won't work.

I'm open to suggestions that would work though. The circular dependencies between the above mentioned classes isn't nice at all. I broke most of them up as best I could, but this local import of Lookup still seems necessary for backwards compatibility.